matrix_sdk/src/models/message.rs

```diff
@@ -0,0 +1,274 @@
+use std::cmp::Ordering;
+use std::ops::Deref;
+use std::vec::IntoIter;
+
+use crate::events::room::message::MessageEvent;
+use crate::events::EventJson;
+
+use serde::{de, ser, Serialize};
+
+/// A queue that holds the 10 most recent messages received from the server.
+#[derive(Clone, Debug, Default)]
+pub struct MessageQueue {
+    msgs: Vec<MessageWrapper>,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct MessageWrapper(MessageEvent);
+
+impl Deref for MessageWrapper {
+    type Target = MessageEvent;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl PartialEq for MessageWrapper {
+    fn eq(&self, other: &MessageWrapper) -> bool {
+        self.0.event_id == other.0.event_id
+            && self.0.room_id == other.0.room_id
+            && self.0.origin_server_ts == other.0.origin_server_ts
+            && self.0.sender == other.0.sender
+            && self.0.content == other.0.content
```

Comment:
This fails now since PartialEq has been removed in Ruma for a bunch of stuff. I think we can just  compare the event ids here, or leave everything besides the content.